### PR TITLE
WIP: add user/message.go 

### PR DESCRIPTION
**Pull request by dingdayu**
I'm trying message.go but I have some trouble and want to set up a discussion.

### CODE REVIEW

File: src/wechaty-puppet/schemas/image.go [hunk 1]
```diff
@@ -3,8 +3,8 @@ package schemas
 type ImageType uint8
 
 const (
-	Unknown   ImageType = 0
-	Thumbnail           = 1
-	HD                  = 2
-	Artwork             = 3
+	ImageTypeUnknown   ImageType = 0
```
Comment (dingdayu) [inline src/wechaty-puppet/schemas/image.go]: Q1: Go does not allow const of the same name and different types in the same package;so I changed the name of the constant, which may differ from the community specification,I wonder if it is appropriate?

File: src/wechaty/accessory.go [hunk 2]
```diff
@@ -4,6 +4,10 @@ import (
 	wechatyPuppet "github.com/wechaty/go-wechaty/src/wechaty-puppet"
 )
 
-type Accessory struct {
-	Puppet wechatyPuppet.Puppet // wechat-puppet 的 Puppet 接口
+type Accessory interface {
```
Comment (dingdayu) [inline src/wechaty/accessory.go]: Q2: I also have some doubts about the definition of Accessory, I am not completely sure.

File: src/wechaty-puppet/puppet.go [hunk 3]
```diff
@@ -1,9 +1,66 @@
 package wechaty_puppet
 
 import (
+	lru "github.com/hashicorp/golang-lru"
+
 	"github.com/wechaty/go-wechaty/src/wechaty-puppet/schemas"
 )
 
-type Puppet interface {
+type PuppetInterface interface {
```
Comment (dingdayu) [inline src/wechaty-puppet/puppet.go]: Q3: In my understanding, Puppet should be an interface definition, different protocols to implement different classes, but some methods of Puppet have logic, and they cannot have logic in Go's interface definition. I don't know what to do.

Sorry, maybe my understanding of wechaty and wechat-puppet is too simple, and I may need to read and understand them as a whole.

**Comment by huan** [comment]
Thanks for your great questions, they are very insightful and valuable!

For better discussion, I have split those three questions into different issues because the PR will be closed and the code reviews will become outdated when we push more code to the PR.

The discussions have been created as #6 (naming style for go) and #7 (accessory questions)

About the Go implementation of the abstract class, I'm not very familiar with the Go specifications, so I'm studying that right now. I Will let you know when I got any ideas about it.

**Comment by huan** [comment]
@SilkageNet Please help us by adding some comments to this PR.

This PR is related to #6 and #7

File: src/wechaty-puppet/puppet.go [hunk 4]
```diff
@@ -1,9 +1,60 @@
...
```
Comment (SilkageNet) [inline src/wechaty-puppet/puppet.go:9]: What about IPuppet?

File: src/wechaty-puppet/schemas/message.go [hunk 5]
```diff
@@ -73,12 +73,16 @@ const (
...
-  Type          MessageType
+	Id string
+
+	// use message id to get rawPayload to get those informations when needed
+	// contactId     string        // Contact ShareCard
+	MentionIdList []string // Mentioned Contacts' Ids
+
+	FileName  string
+	Text      string
+	Timestamp int
```
Comment (SilkageNet) [inline src/wechaty-puppet/schemas/message.go:84]: 我的代码里，timestamp使用的int64，可以考虑一下统一成哪种？

Comment (huan) [reply src/wechaty-puppet/schemas/message.go:84]: Please align with our GRPC service: `uint64`.

See: 

https://github.com/Chatie/grpc/blob/ab85b7345b90c167a031710b9a819f9220c09002/proto/wechaty/puppet/message.proto#L39

Thank you very much.

Comment (SilkageNet) [reply src/wechaty-puppet/schemas/message.go:84]: OK

Comment (dingdayu) [reply src/wechaty-puppet/schemas/message.go:84]: @huan 请教这里的 timestamp 单位是 秒还是毫秒呢？ 

如果是秒的话，我更倾向于 int , 因为当需要转换时会比较方便; 如果是毫秒的话，是需要 int64的。

**Pull request merged**